### PR TITLE
bug fix for https://bugs.launchpad.net/mixxx/+bug/1450391

### PR DESCRIPTION
**Pull request by dodler**
fixed bug 1450391
added some loggng

### CODE REVIEW

**Comment by daschuer** [comment]
The branch has build errors: 

src/widget/wlibrarytableview.cpp:74:59: error: no ‘void WLibraryTableView::setVScrollBarPosState(int)’ member function declared in class ‘WLibraryTableView’

Did you miss to check in a file? 

By the way: I have learned Git by this with fun: 
http://pcottle.github.io/learnGitBranching/

File: src/library/trackmodel.h [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include <QLinkedList>
 #include <QItemDelegate>
 #include <QtSql>
+#include <cmath>
```
Comment (daschuer) [inline src/library/trackmodel.h]: math includes are messed up by the various compilers / libraries. 
We handle this in "util/math.h" please include this instead.

File: src/library/trackmodel.h [hunk 2]
```diff
@@ -21,7 +22,8 @@ class TrackModel {
             : m_db(db),
               m_settingsNamespace(settingsNamespace),
               m_iDefaultSortColumn(-1),
-              m_eDefaultSortOrder(Qt::AscendingOrder) {
+              m_eDefaultSortOrder(Qt::AscendingOrder),
+              position(0){
```
Comment (daschuer) [inline src/library/trackmodel.h]: Nit pick (sorry): Please a a space between ) and { here and else where. 
More infos: http://www.mixxx.org/wiki/doku.php/coding_guidelines

File: src/library/trackmodel.h [hunk 3]
```diff
@@ -144,7 +146,20 @@ class TrackModel {
     virtual void select() {
     }
 
+    virtual int getPosition(){
+      return position;
+    }
+
+    virtual void setPosition(int newPosition){
+      if (newPosition <= 100 && newPosition >= 0 && abs(this->position-newPosition) < 10){
```
Comment (daschuer) [inline src/library/trackmodel.h]: this-> is redundant. Please add a m_ prefix to position to indicate it as member variable.

Comment (daschuer) [inline src/library/trackmodel.h]: Why is this check required? The constants should be used by name like 
const int kFancyPositionChangeLimit = 10;

**Comment by daschuer** [comment]
Thank you very much for adopting this. 

If it is ready, we need you permission to merge. 
Please sign: 
https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
and comment here when done.

**Comment by daschuer** [comment]
Bug #1450391 is a 1.12 Bug.

I can rebase this to 1.2 once it is ready. If you like to do the work, you have to rebase your branch onto 1.12. and you have to open a new PR since GitHub cannot change the target.

**Comment by dodler** [comment]
hello, how to exclude file from commit after pushing it?


>Четверг,  6 августа 2015, 1:20 -07:00 от Daniel Schürmann <notifications@github.com>:
>
>Thank you very much for adopting this.
>If it is ready, we need you permission to merge.
>Please sign:
>https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
>and comment here when done.
>—
>Reply to this email directly or view it on GitHub .


-- 
All the best, 
Artyom Lyan

**Comment by dodler** [comment]
ty for your comments :)

**Comment by daschuer** [comment]
Sorry, it still does not build on Travis and produces some new warnings. 
See https://travis-ci.org/mixxxdj/mixxx/builds/74802328

File: src/library/trackmodel.h [hunk 4]
```diff
@@ -144,7 +147,26 @@ class TrackModel {
...
+      // check of new value can help to prevent such behaviour
+      // better to find why event is emitted
+      if (newPosition <= 100 && newPosition >= 0 && abs(position-newPosition) < positionChangeLimit){
+          position=newPosition;
+        }
+    }
+
   private:
+    // we use it to remember scroll position
+    int position;
```
Comment (daschuer) [inline src/library/trackmodel.h]: please change to 
int m_position

**Comment by dodler** [comment]
merge please

File: src/library/trackmodel.h [hunk 5]
```diff
@@ -15,13 +16,15 @@ class TrackModel {
...
+    static const int positionChangeLimit = 10; // varuable that avoids to too fast scroll
 
     TrackModel(QSqlDatabase db,
                const char* settingsNamespace)
             : m_db(db),
               m_settingsNamespace(settingsNamespace),
               m_iDefaultSortColumn(-1),
-              m_eDefaultSortOrder(Qt::AscendingOrder) {
+              m_eDefaultSortOrder(Qt::AscendingOrder),
+              m_position(0) {
```
Comment (daschuer) [inline src/library/trackmodel.h]: Initalize order does not match the order of the definition in the header. 
You can either mover m_position up here or move the definition to the bottom of the class definition. 

This is the compiler warning: 
```
src/library/trackmodel.h: In constructor ‘TrackModel::TrackModel(QSqlDatabase, const char*)’:
src/library/trackmodel.h:174:19: warning: ‘TrackModel::m_eDefaultSortOrder’ will be initialized after [-Wreorder]
     Qt::SortOrder m_eDefaultSortOrder;
                   ^
src/library/trackmodel.h:168:9: warning:   ‘int TrackModel::m_position’ [-Wreorder]
     int m_position;
         ^
src/library/trackmodel.h:21:5: warning:   when initialized here [-Wreorder]
     TrackModel(QSqlDatabase db,

```

File: src/library/trackmodel.h [hunk 6]
```diff
@@ -144,7 +147,26 @@ class TrackModel {
...
+    }
+
+    virtual void setPosition(int newPosition){
+      // in bug 1450391 we ve got situation
+      // when during changing data model
+      // scrollPositionChanged event is emitted
+      // and scrollBar moves to 15 or 16 from 30-40
+      // check of new value can help to prevent such behaviour
+      // better to find why event is emitted
+      if (newPosition <= 100 && newPosition >= 0 && abs(m_position-newPosition) < positionChangeLimit){
```
Comment (daschuer) [inline src/library/trackmodel.h]: Why newPosition <= 100?

Comment (daschuer) [inline src/library/trackmodel.h]: I have just tried to make this if false by manual test, but I failed. 
What is the receipt for it?

**Comment by daschuer** [comment]
The patch fails, if you move from one crate to an other.

**Comment by rryan** [comment]
Hey @dodler  -- thanks for your contribution! It sounds like there are still issues. Are you still interested in working on this?

**Comment by dodler** [comment]
Hi, yes I'm still interested. Thanks a lot for reminding.

**Comment by Be-ing** [comment]
This pull request has not been updated in years so I am closing it to reduce the clutter of open PRs. If you solve the merge conflicts and would like to have it merged in the future, please reopen it.

**Comment by daschuer** [comment]
This is a progress PR for pending bug 
https://bugs.launchpad.net/mixxx/+bug/1450391
Not sure if this solves the issue entirely though. 
We need to check this. 

I would like to reopen this PR, but it is not possible anymore, because the source repo is deleted.

**Pull request closed without merging**